examples/csv-user-configurations/csv_user_configurations.bal

```diff
@@ -0,0 +1,49 @@
+import ballerina/data.csv;
+import ballerina/io;
+
+// Defines the structure of an `Employee` record.
+type Employee record {|
+    int id;
+    string name;
+    string department;
+|};
+
+// Defines the structure of an `EmployeeDepartment` record.
+type EmployeeDepartment record {|
+    int id;
+    string department;
+|};
+
+public function main() returns error? {
+    // This CSV string contains `|` separated values and a comment.
+    // The `"` character in `O\"Connor` value is escaped using the `\` character.
+    string csvString = string `id|name|department # This is a comment
+                                1|O\"Connor|Engineering
+                                2|Doe|HR
+                                3|Jane|Finance
+                                4|John|Engineering`; 
+                                
+    // Parse the CSV string into an array of `Employee` records with specified options.
+    // The `|` character is used as the delimiter for separating fields.
+    // The `\` character is used to escape characters (e.g., quotes) within the data.
+    // The `#` character indicates the start of a comment, lines starting with `#` will be ignored.
+    // The second and fourth data rows will be skipped during parsing.
+    Employee[] employees = check csv:parseString(csvString, {
+        delimiter: "|",
+        escapeChar: "\\",
+        comment: "#",
+        skipLines: [2, 4]
+    });
```

Comment:
Wouldn't it be better to have the configuration record separately? my thinking is if they want to try out this code they can directly click on the record and go to the definition and explore other config options